pandas/core/generic.py

```diff
@@ -4342,6 +4342,9 @@ def _drop_axis(
                 if errors == "raise" and labels_missing:
                     raise KeyError(f"{labels} not found in axis")
 
+            if is_extension_array_dtype(mask.dtype):
+                mask = mask.to_numpy()
```

Comment:
if mask is BooleanArray, then we need to specify dtype=bool otherwise we get object dtype here.  or could implement EA.nonzero.

comment with GH ref?